client/src/util/stateManager/universe.js

```diff
@@ -0,0 +1,290 @@
+// jshint esversion: 6
+
+import _ from "lodash";
+import KeyValCache from "./keyvalcache";
+
+/*
+This is the public API.   Any non-underscore key in the object is public.
+*/
+class UniverseBase {
+  static VarDataCacheLowWatermark = 32;
+
+  static VarDataCacheTTLMs = 1000;
+
+  /*
+  creates empty universe, noting which REST API version it presumes
+  */
+  constructor(apiVersion) {
+    this.api = apiVersion;
+
+    /* data status */
+    this.finalized = false;
+    this.loading = false;
+    this.error = null;
+
+    this.nObs = 0;
+    this.nVar = 0;
+    this.schema = {};
+
+    /*
+    Annotations
+    */
+    this.obsAnnotations = []; /* all obs annotations, by obs index */
+    this.varAnnotations = []; /* all var annotations, by var index */
+    this.obsNameToIndexMap = {}; /* reverse map 'name' to index */
+    this.varNameToIndexMap = {}; /* reverse map 'name' to index */
+
+    this.obsLayout = { X: [], Y: [] };
+
+    /*
+    Expression cache for var/gene data, keyed by varIndex.  Loaded
```

Comment:
This comment could be expanded. The comment within the keyval cache file is quite extensive - I think it's warranted to speak about the motivations for this function here in narrative form, and the role it plays in the overall system

Reply:
I expanded the comment here and in world.js.   Please review and if acceptable, resolve this comment.   I was not sure exactly what aspects you wanted addressed, so I hit the topics I thought would be most confusing.